test/strings/types.jl

```diff
@@ -26,24 +26,54 @@ for i1 = 1:length(u8str2)
 end
 
 str="tempus fugit"              #length(str)==12
-ss=SubString(str,1,length(str)) #match source string
+ss=SubString(str,1,endof(str)) #match source string
+@test length(ss)==length(str)
+
+ss=SubString(str,1:endof(str))
 @test length(ss)==length(str)
 
 ss=SubString(str,1,0)    #empty SubString
 @test length(ss)==0
 
-ss=SubString(str,14,20)  #start indexed beyond source string length
+ss=SubString(str,1:0)
 @test length(ss)==0
 
-ss=SubString(str,10,16)  #end indexed beyond source string length
-@test length(ss)==3
+# tests for SubString of a single multibyte `Char` string
+# we are consistent with `getindex` for `String`
+for idx in 0:1
+    @test SubString("∀", 1, idx) == "∀"[1:idx]
+end
 
-str2=""
-ss=SubString(str2,1,4)  #empty source string
-@test length(ss)==0
+@test_throws BoundsError SubString("∀", 1, 2)
+@test_throws BoundsError SubString("∀", 1, 3)
+@test_throws BoundsError SubString("∀", 1, 4)
 
-ss=SubString(str2,1,1)  #empty source string, identical start and end index
-@test length(ss)==0
+# tests for SubString of more than one multibyte `Char` string
+# we are consistent with `getindex` for `String`
+for idx in 0:1
+    @test SubString("∀∀", 1, idx) == "∀∀"[1:idx]
+    @test SubString("∀∀", 4, idx) == "∀∀"[4:idx]
+end
+
+@test SubString("∀∀", 4, 4) == "∀∀"[4:4]
+
+for idx in 5:8
```

Comment:
Would be nice to have a comment explaining what is tested here. Same below, where some lines are commented and not others (a global comment for multiple lines is fine).